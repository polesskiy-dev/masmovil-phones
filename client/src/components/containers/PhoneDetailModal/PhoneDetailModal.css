.phone-detail-modal .modal-content {
    background-color: var(--modal-bg-color);
    color: var(--modal-color);
}