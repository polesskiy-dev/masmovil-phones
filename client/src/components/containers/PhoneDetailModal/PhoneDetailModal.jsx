import React from 'react';
import PropTypes from 'prop-types';
import { Modal, ModalHeader, ModalBody } from 'reactstrap';

import './PhoneDetailModal.css';

import ConnectedIntlProvider from '../ConnectedIntlProvider/ConnectedIntlProvider';
import PhoneDetailComponent from '../../presentional/PhoneDetailComponent/PhoneDetailComponent';

const PhoneDetailModal = ({
  modalIsOpen,
  toggleModal,
  selectedDeviceName,
  selectedPhone,
}) => (
  <ConnectedIntlProvider>
    <Modal
      isOpen={modalIsOpen}
      toggle={toggleModal}
      className="phone-detail-modal"
    >
      <ModalHeader toggle={toggleModal}>{selectedDeviceName}</ModalHeader>
      <ModalBody>
        <PhoneDetailComponent {...selectedPhone} />
      </ModalBody>
    </Modal>
  </ConnectedIntlProvider>
);

PhoneDetailModal.propTypes = {
  modalIsOpen: PropTypes.bool.isRequired,
  toggleModal: PropTypes.func.isRequired,
  selectedDeviceName: PropTypes.string.isRequired,
  selectedPhone: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default PhoneDetailModal;
