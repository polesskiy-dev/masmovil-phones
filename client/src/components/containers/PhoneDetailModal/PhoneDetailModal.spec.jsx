import React from 'react';
import renderer from 'react-test-renderer';
import _ from 'lodash/fp';

import withDefaultIntl from '../../../helpers/with-default-intl';
import PhoneDetailModal from './PhoneDetailModal';

const stubPhone = {
  DeviceName: 'Apple iPhone',
  Brand: 'Apple',
  img:
    'https://upload.wikimedia.org/wikipedia/commons/0/02/IPhone_2G_PSD_Mock.png',
  technology: 'GSM',
  announced: '2007, January. Released 2007, June',
  dimensions: '115 x 61 x 11.6 mm (4.53 x 2.40 x 0.46 in)',
  type: 'TFT capacitive touchscreen, 16M colors',
  size: '3.5 inches (~52.0% screen-to-body ratio)',
  os: '',
};

jest.mock(
  '../ConnectedIntlProvider/ConnectedIntlProvider',
  () => ({ children }) => <div>{children}</div>
);

it('<PhoneDetailModal/ > renders correctly when Modal is closed', () => {
  const tree = renderer
    .create(
      withDefaultIntl(
        <PhoneDetailModal
          selectedPhone={stubPhone}
          modalIsOpen={false}
          selectedDeviceName={stubPhone.DeviceName}
          toggleModal={_.noop}
        />
      )
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
