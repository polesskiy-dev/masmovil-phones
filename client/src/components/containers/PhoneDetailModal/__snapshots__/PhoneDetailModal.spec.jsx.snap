// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneDetailModal/ > renders correctly when Modal is closed 1`] = `<div />`;
